### ISSUE TITLE
[calcite-button]: VoiceOver cursor unable to reach button when disabled

### ISSUE THREAD

**Issue by jrowlingson** [Esri/calcite-components#12880]
**monday.com sync:** #18048382042

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Summary

Initially reported [downstream](https://github.com/ArcGIS/field-apps-web/issues/4889).

### Actual Behavior

MacOS VoiceOver cursor is unable to reach `calcite-button` when disabled attribute is set.

### Expected Behavior

MacOS VoiceOver cursor is can reach `calcite-button` when disabled attribute is set.

### Reproduction Sample

https://codepen.io/jrowls/pen/raxBNKg

### Reproduction Steps

https://codepen.io/jrowls/pen/raxBNKg

1. Enable MacOS VoiceOver (VO)
2. Navigate VO cursor to the first `<button>`
3. Advance VO cursor to the second `<button disabled>`
4. Advance VO cursor to the first `<calcite-button>`
5. Attempt to advance VO cursor to the first `<calcite-button disabled>`. Notice VO cursor is unable to reach element.

### Reproduction Version

3.2.1

### Priority impact

impact - p3 - not time sensitive

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Field Apps

**Comment by geospatialem** [comment]
Confirmed the above is scoped to the [above referenced Safari bug](https://github.com/Esri/calcite-design-system/issues/12880#issuecomment-3300121048), and does not impact other browsers or Windows users. Closing the above out as out of scope, as it is related to Safari browser support and outside the scope of Calcite.
